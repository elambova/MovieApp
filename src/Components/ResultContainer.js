import React, { Component } from "react";
import Movies from "./results/Movies";
import People from "./results/People";
import TvShows from "./results/TvShows";

export default class ResultContainer extends Component {
  render() {
    const { data } = this.props;
    const dataLenght = Object.values(data).length;
    const secure_url =
      dataLenght !== 0 ? data.images.secure_base_url + "w185/" : "";

    return (
      <React.Fragment>
        <div className="flex-container">
          {dataLenght > 0 &&
            (() => {
              switch (data.results[0].media_type) {
                case "person":
                  return <People people={data} secure_url={secure_url} />;
                case "movie":
                  return <Movies movies={data} secure_url={secure_url} />;
                case "tv":
                  return <TvShows tv={data} secure_url={secure_url} />;
                default:
                  return null;
              }
            })()}
        </div>
      </React.Fragment>
    );
  }
}
