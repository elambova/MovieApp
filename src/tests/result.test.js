import React from "react";
import Enzyme, { shallow } from "enzyme";
import ResultContainer from "../Components/ResultContainer";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

const data = {
  imagesUrl: "https://image.tmdb.org/t/p/w185",
  apiData: [
    {
      poster_path: "/jGWpG4YhpQwVmjyHEGkxEkeRf0S.jpg",
      popularity: 25.421,
      vote_count: 10081,
      video: false,
      media_type: "movie",
      id: 285,
      adult: false,
      backdrop_path: "/kobscH4qOsYQNTZnkrF5zkPCsvX.jpg",
      original_language: "en",
      original_title: "Pirates of the Caribbean: At World's End",
      genre_ids: [28, 12, 14],
      title: "Pirates of the Caribbean: At World's End",
      vote_average: 7.2,
      overview:
        "Captain Barbossa, long believed to be dead, has come back to life and is headed to the edge of the Earth with Will Turner and Elizabeth Swann. But nothing is quite as it seems.",
      release_date: "2007-05-19",
    },
  ],
};
const handleClickReferrer = () => {
  console.log("handleClickReferrer");
};

const handleClickMovie = () => {
  console.log("handleClickMovie");
};

const handleClickTv = () => {
  console.log("handleClickTv");
};

const handleClickPerson = () => {
  console.log("handleClickPerson");
};

describe("Test ResultContainer", () => {
  let result;
  beforeAll(() => {
    result = shallow(
      <ResultContainer
        data={data}
        handleClickReferrer={handleClickReferrer}
        handleClickMovie={handleClickMovie}
        handleClickTv={handleClickTv}
        handleClickPerson={handleClickPerson}
      />
    );
  });

  it("should be exists", () => {
    expect(result.exists()).toBe(true);
  });

  it("should be render", () => {
    expect(result).toMatchSnapshot();
  });

  it("renders without crashing given the required props", () => {
    expect(toJson(result)).toMatchSnapshot();
  });
});
