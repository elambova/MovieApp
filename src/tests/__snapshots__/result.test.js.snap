// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ResultContainer renders without crashing given the required props 1`] = `
<Fragment>
  <Link
    className="back"
    onClick={[Function]}
    title="Back to Home"
    to="/"
  >
    Â« 
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            576,
            512,
            Array [],
            "f015",
            "M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z",
          ],
          "iconName": "home",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </Link>
  <div
    className="content"
  >
    <Movies
      handleClickMovie={[Function]}
      handleClickTv={[Function]}
      movies={
        Array [
          Object {
            "adult": false,
            "backdrop_path": "/kobscH4qOsYQNTZnkrF5zkPCsvX.jpg",
            "genre_ids": Array [
              28,
              12,
              14,
            ],
            "id": 285,
            "media_type": "movie",
            "original_language": "en",
            "original_title": "Pirates of the Caribbean: At World's End",
            "overview": "Captain Barbossa, long believed to be dead, has come back to life and is headed to the edge of the Earth with Will Turner and Elizabeth Swann. But nothing is quite as it seems.",
            "popularity": 25.421,
            "poster_path": "/jGWpG4YhpQwVmjyHEGkxEkeRf0S.jpg",
            "release_date": "2007-05-19",
            "title": "Pirates of the Caribbean: At World's End",
            "video": false,
            "vote_average": 7.2,
            "vote_count": 10081,
          },
        ]
      }
      secure_url="https://image.tmdb.org/t/p/w185"
    />
  </div>
</Fragment>
`;

exports[`Test ResultContainer should be render 1`] = `ShallowWrapper {}`;
